spec/unit/concerns/connection_spec.rb

```diff
@@ -60,6 +60,22 @@
         end
       end
     end
+
+    describe ":logger" do
+      let(:options) { { adapter: Faraday.default_adapter } }
+
+      before(:each) do
+        client.stub(:authentication_middleware)
+        client.stub(:cache)
+        client.stub(options: options)
+        Restforce.stub(log?: true)
+      end
+
+      it "must always be used last before the Faraday Adapter" do
+        client.middleware.handlers.pop(2).index(Restforce::Middleware::Logger).
```

Comment:
It'd probably be nicer here to avoid `Array#pop` and to more directly check that it's the penultimate one? (e.g. `client.middleware.handlers.reverse.index(Restforce::Middleware::Logger).should eq(1)`)